ext/spl/spl_dllist.c

```diff
@@ -101,7 +101,6 @@ struct _spl_dllist_it {
 	zend_user_iterator     intern;
 	spl_ptr_llist_element *traverse_pointer;
 	int                    traverse_position;
-	int                    flags;
```

Comment:
Based on this struct I'm assuming that SplDoublyLinkedList already allows multiple iteration streams, despite only implementing Iterator? In that case, the current behavior is that calling setIteratorMode() *during* iteration will actually flip direction of all active iterations? Not sure this is the right behavior ... possibly it should be reading from this flags field, rather than the one on the DLL itself.

Reply:
I can investigate further.

Reply:
You are right! I am impressed.

It definitely makes more sense to read from the iterator flags.

Reply:
Let me fix up this PR then...